### PR TITLE
Study coordinator

### PR DESCRIPTION
**Pull request by alxdarksage**

### CODE REVIEW

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/ForStudyCoordinatorsTest.java [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+            coordApi.getStudyParticipantById(STUDY_ID_1, userId, true).execute();
+            fail("Should have thrown exception");
+        } catch(EntityNotFoundException e) {
+            assertEquals("Account not found.", e.getMessage());
+        }
+    }
+    
+    @Test
+    public void createAndThenEnrollStudyParticipant() throws Exception {
+        TestUser researcher = TestUserHelper.createAndSignInUser(ForStudyCoordinatorsTest.class, false, RESEARCHER);
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/ForStudyCoordinatorsTest.java:187]: Cleanup the researcher?

**Review by DwayneJengSage — CHANGES_REQUESTED** [review]
Add tests for following APIs?
* deleteTestParticipant
* getEnrollmentsForUser

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/ForStudyCoordinatorsTest.java [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+
+        RequestInfo info = coordApi.getStudyParticipantRequestInfo(STUDY_ID_1, userId).execute().body();
+        assertNotNull(info);
+        assertEquals("UTC", info.getTimeZone());
+        
+        // Again we just want to see that these don't fail...we can't verify
+        coordApi.sendStudyParticipantEmailVerification(STUDY_ID_1, userId).execute();
+        coordApi.sendStudyParticipantPhoneVerification(STUDY_ID_1, userId).execute();
+        try {
+            coordApi.resendStudyParticipantConsentAgreement(STUDY_ID_1, userId, TEST_APP_ID).execute();
```
[the remark this answers is not in the document]
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/sdk/integration/ForStudyCoordinatorsTest.java:148]: added fail() statements a few places where they were missing. Added tests for deleteTestParticipant and getEnrollmentsForUser. Added consent of user to verify resentStudyParticipantConsentAgreement succeeds.

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/ForStudyCoordinatorsTest.java [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+        coordApi.sendStudyParticipantEmailVerification(STUDY_ID_1, userId).execute();
+        coordApi.sendStudyParticipantPhoneVerification(STUDY_ID_1, userId).execute();
+        try {
+            coordApi.resendStudyParticipantConsentAgreement(STUDY_ID_1, userId, TEST_APP_ID).execute();    
+        } catch(EntityNotFoundException e) {
+            assertEquals("ConsentSignature not found.", e.getMessage());
+        }
+        try {
+            NotificationMessage message = new NotificationMessage().subject("subject").message("message");
+            coordApi.sendStudyParticipantNotification(STUDY_ID_1, userId, message).execute();
```
[the remark this answers is not in the document]
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/sdk/integration/ForStudyCoordinatorsTest.java:154]: added here and elsewhere

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/ForStudyCoordinatorsTest.java [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+        assertEquals(1, list.getItems().size());
+        assertTrue(list.getItems().get(0).isConsentRequired());
+        
+        // get in page
+        AccountSummarySearch search = new AccountSummarySearch().emailFilter(email); 
+        AccountSummaryList summaries = coordApi.getStudyParticipants(STUDY_ID_1, search).execute().body();
+        assertTrue(summaries.getItems().stream().anyMatch(sum -> sum.getId().equals(userId)));
+        
+        try { // this won't work
+            coordApi.getStudyParticipantEnrollments(STUDY_ID_2, userId).execute();
```
[the remark this answers is not in the document]
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/sdk/integration/ForStudyCoordinatorsTest.java:117]: added here and elsewhere

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/ForStudyCoordinatorsTest.java (hunk 1 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/sdk/integration/ForStudyCoordinatorsTest.java:187]: Re-written to clean up account in @AfterTest

**Review by DwayneJengSage — APPROVED** [review]

**Pull request merged**